org.kde.trojita.json

```diff
@@ -0,0 +1,101 @@
+{
+  "app-id": "org.kde.trojita",
+  "runtime": "org.kde.Platform",
+  "runtime-version": "5.15-21.08",
+  "sdk": "org.kde.Sdk",
+  "base": "io.qt.qtwebkit.BaseApp",
+  "base-version": "5.15-21.08",
+  "command": "trojita",
+  "rename-desktop-file": "trojita.desktop",
+  "rename-icon": "trojita",
+  "finish-args": [
+    "--share=ipc",
+    "--share=network",
+    "--socket=wayland",
+    "--socket=fallback-x11",
+    "--device=dri",
+    "--persist=.abook"
+  ],
+  "modules": [
+    {
+      "name": "boost",
+      "buildsystem": "simple",
+      "build-commands": [
+        "./bootstrap.sh --prefix=/app --with-libraries=system",
+        "./b2 -j $FLATPAK_BUILDER_N_JOBS install"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://boostorg.jfrog.io/artifactory/main/release/1.79.0/source/boost_1_79_0.tar.bz2",
+          "sha256": "475d589d51a7f8b3ba2ba4eda022b170e562ca3b760ee922c146b6c65856ef39",
+          "x-checker-data": {
+            "type": "anitya",
+            "project-id": 6845,
+            "stable-only": true,
+            "url-template": "https://boostorg.jfrog.io/artifactory/main/release/$version/source/boost_${major}_${minor}_$patch.tar.bz2"
+          }
+        }
+      ]
+    },
+    {
+      "name": "gpgme",
+      "buildsystem": "cmake-ninja",
+      "config-opts": [
+        "-DBUILD_TESTING=OFF"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "http://download.kde.org/stable/applications/16.08.3/src/gpgmepp-16.08.3.tar.xz",
```

Comment:
gpgmepp is just part of gpgme nowadays, please use https://gnupg.org/ftp/gcrypt/gpgme/gpgme-1.17.1.tar.bz2

Reply:
That might require a patch to be detected:
```
CMake Warning at cmake/TrojitaOption.cmake:121 (find_package):
  By not providing "FindKF5Gpgmepp.cmake" in CMAKE_MODULE_PATH this project                                                                                      
  has asked CMake to find a package configuration file provided by                                                                                               
  "KF5Gpgmepp", but CMake did not find one.                                                                                                                      

  Could not find a package configuration file provided by "KF5Gpgmepp" with                                                                                      
  any of the following names:                                                                                                                                    

    KF5GpgmeppConfig.cmake                                                                                                                                       
    kf5gpgmepp-config.cmake                                                                                                                                      

  Add the installation prefix of "KF5Gpgmepp" to CMAKE_PREFIX_PATH or set                                                                                        
  "KF5Gpgmepp_DIR" to a directory containing one of the above files.  If                                                                                         
  "KF5Gpgmepp" provides a separate development package or SDK, be sure it has                                                                                    
  been installed.                                                                                                                                                
Call Stack (most recent call first):                                                                                                                             
  CMakeLists.txt:97 (trojita_find_package)
```

I'd rather stick to libraries of that era.

Reply:
I'd rather not use unmaintained libraries when it's regarding security stuff like gpg.

Use https://invent.kde.org/pim/trojita/-/commit/e8393376969caf27b0c976d01645edd023c14df2

Reply:
I agree, but
> 1 out of 5 hunks FAILED -- saving rejects to file CMakeLists.txt.rej

Reply:
I was pretty sure I had commented this somewhere already but I can't find the comment anymore.

Yes, you should contact Jan and tell him to do a new release.

Reply:
Ping @jktjkt

Reply:
@Mailaender Thanks for the effort to package Trojita. The best way forward is to stick with any recent git commit; please don't wait for a release.

Reply:
This is not [AUR](https://aur.archlinux.org/packages/trojita-git). The whole Flathub infrastructure relies on releases and monitoring them. Actually, all regular [distributions](https://repology.org/project/trojita/versions) are shipping versions with known vulnerabilities!

Reply:
This is not AUR, but you can use a specific commit just fine.

Reply:
https://dont-ship.it/ though

Reply:
I think the problem is not so much about shipping a commit but about not knowing which one to ship and when to update again. It would be great if the Trojita maintainer could make a new release even a beta one so that we could use a specific release for this Flatpak.